src/flare_pp/kernels/normalized_dot_product.cpp

```diff
@@ -282,6 +282,10 @@ NormalizedDotProduct ::struc_struc(const DescriptorValues &struc1,
         // Energy/energy kernel.
         double norm_dot = dot_vals(i, j) / norm_ij;
         double c1 = (power - 1) * power * pow(norm_dot, power - 2);
+        if (abs(norm_dot) < empty_thresh && power < 2) {
```

Comment:
Do we need `power < 2` in the if statement? Shouldn't this check hold for all powers?

Reply:
Here this one is because c1 is used for stress kernel, and c1 seems to be a 2nd derivative. When the pi and pj are orthogonal and `norm_dot=0`, if `power=1`, then `pow(norm_dot, power - 2) = 1/norm_dot`, which gives a `nan` value